lab_4_retrieval_w_clustering/main.py

```diff
@@ -113,6 +152,23 @@ def put_corpus(self, corpus: Corpus) -> None:
                 or if input arguments are empty,
                 or if methods used return None.
         """
+        if not isinstance(corpus, list) or not corpus:
+            raise ValueError("Input arguments are wrong")
+        tokenized_documents = self._tokenizer.tokenize_documents(corpus)
+        if tokenized_documents is None:
+            raise ValueError("Tokenization returned None")
+
+        filtered_documents = [self._tokenizer._remove_stop_words(paragraph) for paragraph in
+                              tokenized_documents if self._tokenizer._remove_stop_words(paragraph)]
+        if not filtered_documents:
+            raise ValueError("No valid documents after removing stopwords")
+
+        self._vectorizer.set_tokenized_corpus(tokenized_documents)
```

Comment:
for cuurent version of your code, it is incorrect, because you put here `tokenized_documents`, not 'filtered_documents`. But in future, it will be okay

Reply:
it is okay now i believe